### ISSUE TITLE
Fields are sometimes marked as required when they are not

### ISSUE THREAD

**Issue by pablobm** [thoughtbot/administrate#1799]
Merging https://github.com/thoughtbot/administrate/pull/1521 gave Administrate the ability to detect "required" fields and mark them as such.

However, this detection is not always perfect. On occasion, there will be `:if`/`:unless` conditions attached to the validation rule. These are not correctly interepreted by Administrate. So for example:

```rb
  validates :amount, :currency, presence: true, if: ->{ stripe? }
```

<img width="351" alt="Screenshot 2020-10-14 at 18 11 58" src="https://user-images.githubusercontent.com/4217871/96016219-c7f3a600-0e48-11eb-8de9-dfd877de43a5.png">

_Originally posted by @sedubois in https://github.com/thoughtbot/administrate/pull/1521#issuecomment-708506703_

**Comment by jmeinerz** [comment]
@pablobm this should be solved by #1808 (which you could argue open a new issue...)

**Comment by pablobm** [comment]
Fixed by https://github.com/thoughtbot/administrate/pull/1808
